pennylane/gradients/parameter_shift_hessian.py

```diff
@@ -654,6 +654,15 @@ def param_shift_hessian(tape, argnum=None, diagonal_shifts=None, off_diagonal_sh
         )
         return [], lambda _: qml.math.zeros((tape.output_dim, 0, 0))
 
+    # For Jax the argnum defines the tape trainable parameters (it does not work for an array argnum).
+    if (
+        argnum is not None
+        and not isinstance(argnum, np.ndarray)
+        and not tape.trainable_params
+        and qml.math.get_interface(*tape.data) in ["jax", "jax-jit"]
+    ):
+        tape.trainable_params = argnum
+
```

Comment:
I'm a bit worried about this change:
- Newly added similar transforms will have to include this, but it is not clear that this step is necessary, or when this needs to be done.
- It manually tracks trainability, which feels like we should not have to do.

Why is this kind of change not required with gradient transforms? :thinking:

Reply:
I think it is part of a bigger change I want to push in separate PRs, users can use argnums to tag trainable parameters in Jax. Because otherwise everything is considered non trainable when you don't apply gradjac/hessian.